core/src/main/scala/org/apache/spark/eventhubs/client/CachedEventHubsReceiver.scala

```diff
@@ -274,23 +277,27 @@ private[spark] object CachedEventHubsReceiver extends CachedReceiver with Loggin
   private[eventhubs] override def receive(ehConf: EventHubsConf,
                                           nAndP: NameAndPartition,
                                           requestSeqNo: SequenceNumber,
-                                          batchSize: Int): Iterator[EventData] = {
+                                          batchSize: Int,
+                                          eventHubsReceiverListener: Option[EventHubsReceiverListener]
+
+                                         ): Iterator[EventData] = {
     val taskId = EventHubsUtils.getTaskId
 
     logInfo(s"(TID $taskId) EventHubsCachedReceiver look up. For $nAndP, ${ehConf.consumerGroup
       .getOrElse(DefaultConsumerGroup)}. requestSeqNo: $requestSeqNo, batchSize: $batchSize")
     var receiver: CachedEventHubsReceiver = null
     receivers.synchronized {
       receiver = receivers.getOrElseUpdate(key(ehConf, nAndP), {
-        CachedEventHubsReceiver(ehConf, nAndP, requestSeqNo)
+        CachedEventHubsReceiver(ehConf, nAndP, requestSeqNo, eventHubsReceiverListener)
       })
     }
     receiver.receive(requestSeqNo, batchSize)
   }
 
   def apply(ehConf: EventHubsConf,
             nAndP: NameAndPartition,
-            startSeqNo: SequenceNumber): CachedEventHubsReceiver = {
-    new CachedEventHubsReceiver(ehConf, nAndP, startSeqNo)
+            startSeqNo: SequenceNumber,
+            eventHubsReceiverListener: Option[EventHubsReceiverListener]): CachedEventHubsReceiver = {
```

Comment:
eventHubsReceiverListener should have a default None value to ensure backward compatibility